File: examples/14-python-client/producer.py
"""Pure-Python producer for the 14-python-client walkthrough."""
from __future__ import annotations

import argparse

from orion_mesh import Client


def main():
    p = argparse.ArgumentParser()
    p.add_argument("--queue", default="events")
    p.add_argument("--count", type=int, default=20)
    p.add_argument("--prefix", default="hello")
    args = p.parse_args()

    c = Client()
    c.apply(f"""
apiVersion: orionmesh.dev/v1
kind: Queue
metadata: {{ name: {args.queue} }}
spec: {{ type: work, max_age_seconds: 3600 }}
""")
    q = c.queue(args.queue)
    n = q.pub_many({"i": i, "msg": f"{args.prefix}-{i}"} for i in range(args.count))
    print(f"published {n} messages to {q.subject}")
    print(f"queue describe: {c.get('Queue', args.queue).spec}")
    c.close()


if __name__ == "__main__":
    main()

File: examples/15-java-client/src/main/java/io/orionmesh/examples/Producer.java
package io.orionmesh.examples;

import io.orionmesh.client.OrionClient;
import io.orionmesh.client.Queue;

import java.util.HashMap;
import java.util.Map;

public class Producer {
    public static void main(String[] args) throws Exception {
        try (OrionClient c = new OrionClient()) {
            c.apply("""
                apiVersion: orionmesh.dev/v1
                kind: Queue
                metadata: { name: events }
                spec: { type: work, max_age_seconds: 3600 }
                """);
            Queue q = c.queue("events");
            for (int i = 0; i < 20; i++) {
                Map<String, Object> row = new HashMap<>();
                row.put("n", i);
                row.put("msg", "java-" + i);
                long seq = q.pub(row);
                System.out.println("seq=" + seq);
            }
            System.out.println("published 20 messages to " + q.subject());
        }
    }
}
